src/odemis/driver/actuator.py

```diff
@@ -760,10 +760,119 @@ def stop(self, axes=None):
 
     @isasync
     def reference(self, axes):
-        dep_axes = set(self._axes_dep[a] for a in axes)
-        f = self._dependency.reference(dep_axes)
-        return f
+        dep_axes = {self._axes_dep[a] for a in axes}
+        return self._dependency.reference(dep_axes)
+
+class Convert3DStage(ConvertStage):
+    """
+    Extends original ConvertStage with an additional axis Z
+    """
+    def __init__(self, name, role, dependencies, axes,
+                 rotation=(0, 0, 0), scale=None, translation=None, **kwargs):
+        """
+        dependencies (dict str -> actuator): name to objective lens actuator
+        axes (list of 3 strings): names of the axes for x, y and z
+        scale (None tuple of 3 floats): scale factor from exported to original position
+        rotation (tuple of 3 floats): rotation factor rz, ry, rx (in radians)
+        translation (None or tuple of 3 floats): translation offset (in m)
+        """
+        assert len(axes) == 3
+        if len(dependencies) != 1:
+            raise ValueError("ConvertStage needs 1 dependency")
+
+        self._dependency = list(dependencies.values())[0]
+        self._axes_dep = {"x": axes[0], "y": axes[1], "z": axes[2]}
+        if scale is None:
+            scale = (1, 1, 1)
+        if translation is None:
+            translation = (0, 0, 0)
```

Comment:
Why not directly indicating these default values in the signature too? As a tuple cannot be changed, it should be safe.